keps/sig-apps/3329-retriable-and-non-retriable-failures/README.md

```diff
@@ -409,6 +433,52 @@ Note that, in this case the user supplies a list of Pod condition type values.
 This approach is likely to require an iterative process to review and extend of
 the list.
 
+#### Story 3
+
+As a service provider, similarly as in [Story 2](#story-2), I would like to
+have a mechanism which terminates jobs for which pods are failing due to user
+errors. However, I'm concerned that jobs running on a cluster with a broken
+configuration might result in costly never-ending Pod retries.
+
+The following Job configuration could be a good starting point to satisfy
+my needs:
+
+```yaml
+apiVersion: v1
+kind: Job
+spec:
+  template:
+    spec:
+      containers:
+      - name: main-job-container
+        image: job-image
+        command: ["./program"]
+        resources:
+          requests:
+            memory: "10Gi"
+            ephemeral-storage: "10Gi"
+          limits:
+            memory: "10Gi"
+            ephemeral-storage: "100Gi"
+  backoffLimit: 8
+  podFailurePolicy:
```

Comment:
I wonder how the complete list of rules will look in a "best practice" job. Will there ever be the case when `ResourceExhausted` will be retrieable?

Reply:
I think the “best practice” configuration would be similar to this one (Story 3). 

Generally you want to fail the job if `ResourceExhausted` (as in the example). The only reason would want to retry on `ResourceExhausted` is if the Pod was terminated by OOM killer invoked due to the node running low on memory. Note that, OOM killer invoked due to actual exceeding memory limits by a container is fine and should trigger `FailJob` as it indicates there is a bug (memory leak) or job misconfiguration. Unfortunately, there is currently no way to distinguish the two situations (OOM due to exceeding limits and OOM due to the node running low on memory). We will recommend users to minimize the risk by:
- job configuration specifying requests.memory=limits.memory for all containers
- leaving enough available memory and reserved for the system, see: https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/#node-pressure-eviction-good-practices

Reply:
Also, rephrased the Risks and Mitigations section to include the second point about kubelet configuration

Reply:
> job configuration specifying requests.memory=limits.memory for all containers

should we then use the warnings mechanism to warn users that when they use failure policy those two should match?

Reply:
I think this is a sensible idea, but I would restrict the `requests.memory=limits.memory` check only when there is a rule with `ResourceExhausted` as condition type and `FailJob` as action. 

What's your view @alculquicondor @soltysh ?

Reply:
SGTM

Reply:
Added a paragraph to the  Risks and Mitigations section